presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -820,7 +820,7 @@ private static Table buildTableObject(
             Optional<HiveBucketProperty> bucketProperty,
             Map<String, String> additionalTableParameters,
             Path targetPath,
-            boolean external,
+            String tableType,
```

Comment:
nit: Why pass in `String` instead of `TableType`?

Reply:
I see the reason , in "Implement temporary tables in Hive connector" you are going to introduce `__PRESTO_TEMPORARY_TABLE__`.

What about have own own `PrestoTableType` enum? It can be `HIVE_MANAGED_TABLE`, `HIVE_EXTERNAL_TABLE` and `PRESTO_TEMPORARY_TABLE` :)

Reply:
The `tableType` filed was deliberately left as string: https://github.com/prestodb/presto/blob/95c8715fa4edcabba85b0f7edae14294abb78a59/presto-hive/src/main/java/com/facebook/presto/hive/metastore/Table.java#L41

I'm not sure if it is worth it to introduce a special enum just to pass it to this private method.

Reply:
@arhimondr 

> The tableType filed was deliberately left as string

hmm. I see. In that case we should have comments about known possible values :)

Reply:
Actually let me introduce simple enum that is local to the `HiveMetadata` just for convinience. I don't think there's anything bad about it, even if we won't pass it to the `Table` object itself.

Reply:
Actually i decided to just introduce an enum, `PrestoTableType`

Reply:
I added `PrestoTableType` enum, and refactored the `Table` class to use that enum. Had to add additional workaround for the Glue metastore implementation to handle `get-update-set` cases correctly.